### DISCUSSION TITLE
Pods of Crossplane are in "CrashLoopBackOff" when we restore CrossPlane using Velero

### DISCUSSION THREAD

**Discussion by Harshith-Krishnappa96** [vmware-tanzu/velero#7075] [Community support Q&A]
Hi Team,

I have taken the backup of "crossplane" namespace using velero in EKS Cluster1 and want to restore it in EKS Cluster2 in the same region.
When I try to restore below is the output:

QPHXH6KFPV:~ harshith.gowda$ velero restore describe cross-plane-backup-20231108073039-20231108130603
Name:         cross-plane-backup-20231108073039-20231108130603
Namespace:    velero
Labels:       <none>
Annotations:  <none>

Phase:                       Completed
Total items to be restored:  17
Items restored:              17

Started:    2023-11-08 13:06:06 +0530 IST
Completed:  2023-11-08 13:06:11 +0530 IST

Warnings:
  Velero:     <none>
  Cluster:    <none>
  Namespaces:
    crossplane-system:  could not restore, ConfigMap "kube-root-ca.crt" already exists. Warning: the in-cluster version is different than the backed-up version

Backup:  cross-plane-backup-20231108073039

Namespaces:
  Included:  crossplane-system
  Excluded:  <none>

Resources:
  Included:        *
  Excluded:        nodes, events, events.events.k8s.io, backups.velero.io, restores.velero.io, resticrepositories.velero.io, csinodes.storage.k8s.io, volumeattachments.storage.k8s.io, backuprepositories.velero.io
  Cluster-scoped:  auto

Namespace mappings:  <none>

Label selector:  <none>

Or label selector:  <none>

Restore PVs:  auto

Existing Resource Policy:   <none>
ItemOperationTimeout:       4h0m0s

Preserve Service NodePorts:  auto

What am I missing as it says:     "crossplane-system:  could not restore, ConfigMap "kube-root-ca.crt" already exists. Warning: the in-cluster version is different than the backed-up version"

Even though the restoration seems to be completed but the pods of the CrossPlane is facing below issue:

QPHXH6KFPV:~ harshith.gowda$ kubectl get pods -n crossplane-system
NAME                                       READY   STATUS                  RESTARTS      AGE
crossplane-747b75765f-rtwmm                0/1     Init:CrashLoopBackOff   3 (17s ago)   67s
crossplane-rbac-manager-5c9c4f777f-q6lmk   0/1     Init:Error              3 (44s ago)   67s

And when I describe the pods below is the event:
Events:
  Type     Reason     Age                  From               Message
  ----     ------     ----                 ----               -------
  Normal   Scheduled  12m                  default-scheduler  Successfully assigned crossplane-system/crossplane-rbac-manager-5c9c4f777f-q6lmk to ip-10-0-0-44.us-west-2.compute.internal
  Normal   Pulling    12m                  kubelet            Pulling image "crossplane/crossplane:v1.11.1"
  Normal   Pulled     12m                  kubelet            Successfully pulled image "crossplane/crossplane:v1.11.1" in 2.696331555s (2.696348958s including waiting)
  Normal   Created    10m (x5 over 12m)    kubelet            Created container crossplane-init
  Normal   Started    10m (x5 over 12m)    kubelet            Started container crossplane-init
  Normal   Pulled     10m (x4 over 12m)    kubelet            Container image "crossplane/crossplane:v1.11.1" already present on machine
  Warning  BackOff    2m8s (x46 over 12m)  kubelet            Back-off restarting failed container crossplane-init in pod crossplane-rbac-manager-5c9c4f777f-q6lmk_crossplane-system(04eec59c-8a9c-419d-9717-c33dd1fcc

Please help me in advising how to overcome this issue

**Comment by sseago** [comment]
CrashLoopBackOff means that the pod is starting successfully, but then repeatedly crashing. Next step would be to examine pod logs -- there's a good chance you'll see error messages of some sort there. Also, check the pod description -- make sure everything is correct there in terms of container definition, environment, etc.

**Comment by Harshith-Krishnappa96** [comment]
Hi @sseago : Thanks for your response. I get the below response when I describe the pods:

kubelet Container image "crossplane/crossplane:v1.11.1" already present on machine
kubelet Back-off restarting failed container crossplane-init in pod crossplane-rbac-manager-5c9c4f777f-q6lmk_crossplane-system(04eec59c-8a9c-419d-9717-c33dd1fcc

**Comment by sseago** [comment]
@Harshith-Krishnappa96 So it would appear that the pod is crashing. That's all the event messages are telling you. To learn more, you probably need to look at the pod logs -- that may have actual application logs or segfault stacktraces or some other information as to what's going wrong in your application. If nothing there, look at your pod spec to see if there's anything wrong, perhaps a misconfiguration etc.
